app/src/main/java/com/provider/persistence/entity/Provider.java

```diff
@@ -53,7 +55,8 @@ public class Provider {
   @UpdateTimestamp
   private Timestamp updatedAt;
 
-  @OneToMany(mappedBy = "provider", cascade = CascadeType.PERSIST)
+  @OneToMany(mappedBy = "provider", cascade = CascadeType.PERSIST, fetch = FetchType.EAGER)
+  @Where(clause = "status != 'cancelled'")
```

Comment:
Not this way.
This will work, but it will restrict your ability to get all items when there is such a need.
Please, use a custom JPA query on the repository level.